backend/src/main/java/gov/cdc/usds/simplereport/db/model/ConsoleApiAuditEvent.java

```diff
@@ -0,0 +1,78 @@
+package gov.cdc.usds.simplereport.db.model;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import gov.cdc.usds.simplereport.config.authorization.UserPermission;
+import gov.cdc.usds.simplereport.db.model.auxiliary.GraphQlInputs;
+import gov.cdc.usds.simplereport.db.model.auxiliary.HttpRequestDetails;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import org.apache.http.HttpStatus;
+
+@Getter
+public class ConsoleApiAuditEvent {
+  private final HttpRequestDetails httpRequestDetails;
+  private static final String TYPE = "auditLog";
+  private GraphQlInputs graphqlQueryDetails;
+  private List<String> graphqlErrorPaths;
+  private List<String> userPermissions;
+  private Organization organization;
+  private PatientLink patientLink;
+  private final String requestId;
+  private final int responseCode;
+  private boolean isAdminUser;
+  private final ApiUser user;
+  private JsonNode session;
+
+  /** Constructor for graphql requests */
+  public ConsoleApiAuditEvent(
+      String requestId,
+      HttpRequestDetails httpRequestDetails,
+      GraphQlInputs graphqlQueryDetails,
+      List<String> errorPaths,
+      ApiUser apiUser,
+      List<UserPermission> permissions,
+      boolean isAdmin,
+      Organization org) {
+    this.responseCode = HttpStatus.SC_OK;
+    this.graphqlQueryDetails = graphqlQueryDetails;
+    this.graphqlErrorPaths = errorPaths;
+    this.userPermissions =
+        permissions.stream().map(UserPermission::name).sorted().collect(Collectors.toList());
+    this.isAdminUser = isAdmin;
+    this.organization = org;
+    this.httpRequestDetails = httpRequestDetails;
+    this.requestId = requestId;
+    this.user = apiUser;
+  }
+
+  /** Constructor for REST (patient-experience) requests */
+  public ConsoleApiAuditEvent(
+      String requestId,
+      HttpRequestDetails httpRequestDetails,
+      int responseStatus,
+      ApiUser user,
+      Organization organization,
+      PatientLink patientLink) {
+    this.patientLink = patientLink;
+    this.organization = organization;
+    this.user = user;
+    this.responseCode = responseStatus;
+    this.httpRequestDetails = httpRequestDetails;
+    this.requestId = requestId;
+  }
+
+  /** Constructor for anonymous REST requests. */
+  public ConsoleApiAuditEvent(
```

Comment:
Note that I don't think we actually use this endpoint at the moment. I'd still like to move it over though, there are definitely some anonymous requests I'd like audit logging on if possible.

Reply:
yup! Hopefully this will be cleaned up when the cleanup happens for the code for registration before test flow is removed